app/src/main/java/de/danoeh/antennapod/activity/MainActivity.java

```diff
@@ -323,6 +327,34 @@ public void onConfigurationChanged(Configuration newConfig) {
         if (drawerToggle != null) { // Tablet layout does not have a drawer
             drawerToggle.onConfigurationChanged(newConfig);
         }
+        setNavDrawerSize();
+    }
+
+    private void setNavDrawerSize() {
+        float screenPercent = getResources().getInteger(R.integer.nav_drawer_screen_size_percent) * 0.01f;
+        int width = (int) (getScreenWidth() * screenPercent);
+        int maxWidth = (int) dpInPx(getResources().getInteger(R.integer.nav_drawer_max_screen_size_dp));
+
+        if (width > maxWidth) {
+            width = maxWidth;
```

Comment:
Android Lint does not like manual `min` calculation. Something like `Math.min()` would be more clear.